eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -723,6 +777,7 @@ void register() {
     @PreDestroy
     @Override
     public void shutdown() {
+        ApplicationInfoManager.getInstance().unregisterStatusChangeListener(statusChangeListener.getId());
```

Comment:
Is it possible to pass ApplicationInfoManager instance in constructor via injection to eliminate this singleton access?